sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextBase.java

```diff
@@ -218,6 +219,18 @@ public CosmosQueryRequestOptions getFeedOptions(String continuationToken, Intege
             requestHeaders.put(HttpConstants.HttpHeaders.POPULATE_QUERY_METRICS, String.valueOf(cosmosQueryRequestOptions.isQueryMetricsEnabled()));
         }
 
+        if (cosmosQueryRequestOptions.getDedicatedGatewayRequestOptions() != null &&
+            cosmosQueryRequestOptions.getDedicatedGatewayRequestOptions().getMaxIntegratedCacheStaleness() != null) {
+            Duration maxIntegratedCacheStaleness = cosmosQueryRequestOptions
+                .getDedicatedGatewayRequestOptions()
+                .getMaxIntegratedCacheStaleness();
+            if (maxIntegratedCacheStaleness.toNanos() > 0 && maxIntegratedCacheStaleness.toMillis() <= 0) {
```

Comment:
Check if maxIntegratedCacheStaleness is negative

Reply:
Added the check.